specifications/cloud_assembly.md

```diff
@@ -0,0 +1,398 @@
+
+## Annex
+### Examples of Drops for the AWS Cloud
+#### `@aws-cdk/aws-cloudformation.StackDrop`
+A [*CloudFormation* stack][CFN Stack].
+
+##### Properties
+Property     |Type                |Required|Description
+-------------|--------------------|:------:|-----------
+`template`   |`string`            |Required|The assembly-relative path to the *CloudFormation* template document.
+`parameters` |`Map<string,string>`|        |Parameters to be passed to the [stack][CFN Stack] upon deployment.
+`stackPolicy`|`string`            |        |The assembly-relative path to the [Stack Policy][CFN Stack Policy].
+
+##### Output Attributes
+Attribute |Type                |Description
+----------|--------------------|-----------
+`outputs` |`Map<string,string>`|Data returned by [*CloudFormation* Outputs][CFN Outputs] of the stack.
+`stackArn`|`string`            |The ARN of the [stack][CFN Stack].
+
+##### Example
+```json
+{
+  "type": "npm://@aws-cdk/aws-cloudformation.StackDrop",
+  "environment": "aws://000000000000/bermuda-triangle-1",
+  "properties": {
+    "template": "my-stack/template.yml",
+    "parameters": {
+      "bucketName": "${helperStack.bucketName}",
+      "objectKey": "${helperStack.objectKey}"
+    },
+    "stackPolicy": "my-stack/policy.json"
+  }
+}
+```
+
+[CFN Stack]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stacks.html
+[CFN Stack Policy]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/protect-stack-resources.html
+[CFN Outputs]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/outputs-section-structure.html
+
+#### `@aws-cdk/assets.FileDrop`
+A file that needs to be uploaded to an *S3* bucket.
+
+##### Properties
+Property    |Type    |Required|Description
+------------|--------|:------:|-----------
+`file`      |`string`|Required|The assembly-relative path to the file that will be uploaded.
+`bucketName`|`string`|Required|The name of the bucket where this file will be uploaded.
+`objectKey` |`string`|Required|The key at which to place the object in the bucket.
+
+##### Output Attributes
+Attribute   |Type    |Description
+------------|--------|-----------
+`bucketName`|`string`|The name of the bucket where the file was uploaded.
+`objectKey` |`string`|The key at which the file was uploaded in the bucket.
+
+##### Example
+```json
+{
+  "type": "npm://@aws-cdk/assets.FileDrop",
+  "environment": "aws://000000000000/bermuda-triangle-1",
+  "properties": {
+    "file": "assets/file.bin",
+    "bucket": "${helperStack.bucketName}",
+    "objectKey": "assets/da39a3ee5e6b4b0d3255bfef95601890afd80709/nifty-asset.png"
+  }
+}
+```
+
+#### `@aws-cdk/aws-ecr.DockerImageDrop`
+A Docker image to be published to an *ECR* registry.
+
+##### Properties
+Property    |Type    |Required|Description
+------------|--------|:------:|-----------
+`savedImage`|`string`|Required|The assembly-relative path to the tar archive obtained from `docker image save`.
```

Comment:
Can we also have a `localImageName` for a simplified workflow of `docker build + docker push`, instead of a `docker build + docker save + docker push`?

I'd like to save copying 1GB of data around on every build if it's not necessary.

Reply:
Added a short blurb to note that the AWS Drop specifications provided in the document are for illustration purpose only. The API design for those can be a separate process :) But I hear the concern and it's valid, I just don't want to overload the Cloud Assembly spec with those details (at least not right now).